juju/juju
# Implement the consume command using the new application offer APIs

## Description of change

Add the controller facade GetConsumeDetails API. This allows the consume command to be re-implemented and functional again. As part of the work, a whole lot of code was consolidated into fewer common methods and a hundreds of lines deleted. The core facade apis remain the same, with the addition of the new api mentioned above.

For future use, the api params also cater for handling a macaroon which will be used for authentication between controllers/models.

## QA steps

juju bootstrap
juju deploy mysql
juju offer mysql:db
juu switch controller
juju consume default.mysql
juju status

Also check the operation of the other cmr commands find-endpoints etc
